WeakAuras/Transmission.lua

```diff
@@ -972,13 +1026,81 @@ function WeakAuras.ShowDisplayTooltip(data, children, icon, icons, import, compr
           local co = coroutine.create(func);
           WeakAuras.dynFrame:AddAction("import", co);
         end);
+        if matchedData then
+          updatebutton:SetText("Update")
+          updatebutton:SetScript("OnClick", function()
+            local func = function()
+              WeakAuras.SetImporting(true)
+              WeakAuras.LoadOptions()
+
+              local optionsFrame = WeakAuras.OptionsFrame();
+              if not(optionsFrame) then
+                WeakAuras.ToggleOptions();
+                optionsFrame = WeakAuras.OptionsFrame();
+              end
+
+              if not(WeakAuras.IsOptionsOpen()) then
+                WeakAuras.ToggleOptions();
+              end
+
+              local function importUpdatedData(data, oldData)
+                if oldData then
+                  WeakAuras.Update(data, oldData, "important");
```

Comment:
I'm a bit suprised that the modified check and the importing aren't symetric. That is a aura can be modifed, yet the importing doesn't actually update anything. I would have expected WeakAuras.Update and WeakAuras.diff to use the same criteria.

Reply:
Update could, depending on user preferences, be less zealous than Diff. Diff is intended to find the differences (mainly so that an unmonitored version can be detected and skipped in the update process), while Update is supposed to (in a future iteration) change behavior based on user preferences.

Reply:
Well, in the future we'll solve all kinds of problems. But not updating a change that diff detects sounds wrong to me.